src/s3ql/backends/gdrive.py

```diff
@@ -0,0 +1,599 @@
+'''
+backends/gdrive.py - this file is part of S3QL.
+
+Copyright © 2017 Isaac Aymerich <isaac.aymerich@gmail.com>
+
+This work can be distributed under the terms of the GNU GPLv3.
+'''
+
+
+from ..logging import logging # Ensure use of custom logger class
+from ..inherit_docstrings import (copy_ancestor_docstring,prepend_ancestor_docstring, ABCDocstMeta)
+from .common import (AbstractBackend, DanglingStorageURLError, NoSuchObject,retry,retry_generator,AuthenticationError)
+import apiclient.discovery
+import apiclient.http
+import time
+import base64
+import hashlib
+import httplib2
+import ssl
+import http
+import socket
+import io
+import json
+import oauth2client.client
```

Comment:
Is this really required? The google storage backend uses OAuth just fine without extra modules.

Reply:
this imports are because the function is_temp_failure to check if the error is one of them. We should wrapp all the errors in a generic Exception probably

Reply:
I'm talking specifically about `oauth2client.client`. Just like with `google-api-python-client` I am worried that it may be doing much more than we want. To clarify: if omitting this dependency means hundreds of lines of extra code we should of course use the module. But if a from scratch implementation of the subset of functions that we require for S3QL is just a hundred lines or so, then I wouldn't want to use an external module for that.